src/libcrun/container.c

```diff
@@ -1187,6 +1187,13 @@ container_init_setup (void *args, pid_t own_pid, char *notify_socket, int sync_s
   if (clearenv ())
     return crun_make_error (err, errno, "clearenv");
 
+  // set primary process to 1 explicitly if nothing is configured and LISTEN_FD is not set
+  if (entrypoint_args->context->listen_fds > 0)
+    {
+      setenv ("LISTEN_PID", "1", 1);
+      libcrun_warning ("setting LISTEN_PID=1 since no previous configuration was found");
```

Comment:
don't we need to make sure the variable wasn't already set like: `if (entrypoint_args->context->listen_fds > 0 &&  getenv ("LISTEN_PID") == NULL)`?

I think at this point it is always unset, since we are calling it just after `clearenv ()`.  We need to make sure this code snippet is done below the other `putenv` and `setenv`.

I think this can be easily tested, and we can add a test for it.

Reply:
Yes it has to be below `putenv(def->process->env[i])` and `set_home_env` making changes and adding tests as well.